app/lib/util/native/file_picker.dart

```diff
@@ -225,48 +240,78 @@ Future<void> _pickFolder(BuildContext context, Ref ref) async {
 
     _logger.warning('Failed to pick directory', e);
     // ignore: use_build_context_synchronously
-    await showDialog(context: context, builder: (_) => const NoPermissionDialog());
+    await showDialog(
+        context: context, builder: (_) => const NoPermissionDialog());
   } finally {
     // ignore: use_build_context_synchronously
     Routerino.context.popUntilRoot(); // remove loading dialog
   }
 }
 
 Future<void> _pickMedia(BuildContext context, Ref ref) async {
-  final oldBrightness = Theme.of(context).brightness;
-  // ignore: use_build_context_synchronously
-  final List<AssetEntity>? result = await AssetPicker.pickAssets(
-    context,
-    pickerConfig: const AssetPickerConfig(maxAssets: 999, textDelegate: TranslatedAssetPickerTextDelegate()),
-  );
+  if (ref.read(settingsProvider).privacyProtectionMode) {
+    final ImagePicker picker = ImagePicker();
+    final List<XFile>? images = await picker.pickMultiImage(requestFullMetadata:false);
+    if (images != null && images.isNotEmpty) {
+      // 将文件转换器函数赋值给一个变量
+      final converterFunction = CrossFileConverters.convertXFile;
 
-  WidgetsBinding.instance.addPostFrameCallback((_) async {
-    // restore brightness for Android
-    await sleepAsync(500);
-    if (context.mounted) {
-      await updateSystemOverlayStyleWithBrightness(oldBrightness);
-    }
-  });
+      // 分发文件添加的 action，并使用定义的转换器
+      await ref
+          .redux(selectedSendingFilesProvider)
+          .dispatchAsync(AddFilesAction(
+            files: images,
+            converter: converterFunction,
+          ));
+    } else {}
+  } else {
+    final oldBrightness = Theme.of(context).brightness;
+    // ignore: use_build_context_synchronously
+    final List<AssetEntity>? result = await AssetPicker.pickAssets(
+      context,
+      pickerConfig: const AssetPickerConfig(
+          maxAssets: 999, textDelegate: TranslatedAssetPickerTextDelegate()),
+    );
 
-  if (result != null) {
-    await ref.redux(selectedSendingFilesProvider).dispatchAsync(AddFilesAction(
-          files: result,
-          converter: CrossFileConverters.convertAssetEntity,
-        ));
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      // restore brightness for Android
+      await sleepAsync(500);
+      if (context.mounted) {
+        await updateSystemOverlayStyleWithBrightness(oldBrightness);
+      }
+    });
+
+    if (result != null) {
+      // 将文件转换器函数赋值给一个变量
```

Comment:
English is the development language only. Do remove these comments.

Reply:
comments removed